corpus/pages/views.py

```diff
@@ -26,9 +28,24 @@ def about_us(request):
         },
     )
 
+
 def impulse(request):
 
     return render(
         request,
         "pages/impulse.html",
-    )
\ No newline at end of file
+    )
+
+
+def sig(request, sig_slug):
+    sig_data = get_object_or_404(SIG, slug=sig_slug)
+
+    # Retrieve the related society details using the SIG instance
+    societies_linked_to_sig = sig_data.societies.all()
+
+    args = {
```

Comment:
Lets keep `render` definitions consistent. Since we've defined the context dict for `render` directly as a `dict` in the function arguments for the other views, lets not define a separate `args` variable and reference it in the function call.

This `dict` can be defined in the function call itself. See other views in this file for reference (example, `index` and `about_us`).

Reply:
Oh alright, actually @anirudhprabhakaran3  had followed using args in the embedathon app, so I was following that format.
But even this is alright, I've reverted my change

Reply:
thanks!